tests/test_utils_gz.py

```diff
@@ -25,5 +25,10 @@ def test_gunzip_no_gzip_file_raises(self):
 
     def test_gunzip_truncated_short(self):
         with open(join(SAMPLEDIR, 'truncated-crc-error-short.gz'), 'rb') as f:
-            text = gunzip(f.read())
+            text = gunzip(f.read()) 
             assert text.endswith(b'</html>')
+
+    def test_gunzip_illegal_eof(self):
+        with open(join(SAMPLEDIR, 'unexpected-eof.gz'), 'rb') as f:
+            text = gunzip(f.read())
+            self.assertEqual(text.count(b'document'), 1)
```

Comment:
Another nitpick also ;) We try to follow PEP8 in Scrapy's codebase, please end the file (meaning `test_utils_gz.py` here) in a newline (i.e. the file ends with `\n`). Note that this does not mean that there should be an empty line (which would be `\n\n`). You can probably configure your editor to do it automatically for you (iirc vim does it by default).